graphbolt/src/cuda/max_uva_threads.h

```diff
@@ -0,0 +1,24 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ *  Copyright (c) 2023, GT-TDAlab (Muhammed Fatih Balin & Umit V. Catalyurek)
+ * @file cuda/max_uva_threads.h
+ * @brief Max uva threads variable declaration.
+ */
+#ifndef GRAPHBOLT_MAX_UVA_THREADS_H_
+#define GRAPHBOLT_MAX_UVA_THREADS_H_
+
+#include <cstdint>
+#include <optional>
+
+namespace graphbolt {
+namespace cuda {
+
+/** @brief Set a limit on the number of CUDA threads for UVA accesses. */
+inline std::optional<int64_t> max_uva_threads;
```

Comment:
I don't really understand the keyword inline, for defining global variables should we use `extern std::optional<int64_t> max_uva_threads;`?

Reply:
using inline in the header file ensures that the linker only keeps one instantiation of the variable even if the header is included in multiple source files.

Reply:
That is what I learned from the internet from my brief research.

Reply:
I tested the code, it works even though multiple source files include this header.

Reply:
https://en.cppreference.com/w/cpp/language/inline, Explanation section. It looks like it is a C++17 feature.